File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);


int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    float L = (float)letters / words * 100;
    float S = (float)sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int roundedIndex = (int)round(index);

    if (roundedIndex < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (roundedIndex >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }

    printf("Grade %d\n", roundedIndex);
    return 0;
}

int count_letters(string text)
{
    int counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        char letter = tolower(text[i]);
        if ((int)letter - 97 >= 0 && (int)letter - 97 <= 25)
        {
            counter++;
        }
    }

    return counter;
}

int count_words(string text)
{
    int counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        char letter = tolower(text[i]);
        if ((int)letter - 32 == 0)
        {
            counter++;
        }
    }

    return counter + 1;
}

int count_sentences(string text)
{
    int counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        char letter = tolower(text[i]);
        if ((int)letter - 46 == 0 || (int)letter - 33 == 0 || (int)letter - 63 == 0)
        {
            counter++;
        }
    }

    return counter;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():

    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    roundedIndex = int(round(index))

    if roundedIndex < 1:
        print("Before Grade 1")
        return 0
    elif roundedIndex >= 16:
        print("Grade 16+")
        return 0

    print(f"Grade {roundedIndex}")
    return 0


def count_letters(text):
    counter = 0
    for i in range(len(text)):
        letter = text[i].lower()
        if ord(letter) - 97 >= 0 and ord(letter) - 97 <= 25:
            counter += 1

    return counter


def count_words(text):
    counter = 0
    for i in range(len(text)):
        letter = text[i].lower()
        if ord(letter) - 32 == 0:
            counter += 1

    return counter + 1


def count_sentences(text):
    counter = 0
    for i in range(len(text)):
        letter = text[i].lower()
        if ord(letter) - 46 == 0 or ord(letter) - 33 == 0 or ord(letter) - 63 == 0:
            counter += 1

    return counter


main()